### PR TITLE
Sudt support

### PR DESCRIPTION
**Pull request by classicalliu**

### CODE REVIEW

File: packages/common-scripts/src/common.ts [hunk 1]
```diff
@@ -86,27 +88,27 @@ export async function transfer(
   if (useLocktimeCellsFirst) {
     // if provider tipHeader
     if (tipHeader) {
-      [txSkeleton, deductAmount] = await lockTimePool.transfer(
+      const result = await lockTimePool.transfer(
```
Comment (xxuejie) [inline packages/common-scripts/src/common.ts]: Any reason for this change?

File: packages/common-scripts/src/sudt.ts [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+  parseAddress,
+  minimalCellCapacity,
+  TransactionSkeletonType,
+  Options,
+} from "@ckb-lumos/helpers";
+import { Set, List } from "immutable";
+import { getConfig, Config } from "@ckb-lumos/config-manager";
+import { collectCells, LocktimeCell } from "./locktime_pool";
+
+export async function createToken(
```
Comment (xxuejie) [inline packages/common-scripts/src/sudt.ts]: issueToken is probably a better name.

File: packages/common-scripts/src/sudt.ts [hunk 3]
```diff
@@ -0,0 +1,421 @@
...
+ * @param toAddress
+ * @param amount
+ * @param changeAddress if not provided, will use first fromInfo
+ * @param capacity
+ * @param options
+ */
+export async function transfer(
+  txSkeleton: TransactionSkeletonType,
+  fromInfos: FromInfo[],
+  sudtToken: Hash,
```
Comment (xxuejie) [inline packages/common-scripts/src/sudt.ts]: Let's define a `Token` type, for now it could be just the args, but later we can expand this to full Script

**Review by xxuejie — COMMENTED** [review]
Let's also add a short introduction on how to use it in the README part.

File: packages/common-scripts/src/common.ts (hunk 1 above)
Comment (classicalliu) [reply packages/common-scripts/src/common.ts]: I forget to revert this after debug, I'll revert changes in `transfer`, it's no need to change now.

File: packages/common-scripts/src/sudt.ts (hunk 2 above)
Comment (classicalliu) [reply packages/common-scripts/src/sudt.ts]: updated.

File: packages/common-scripts/src/sudt.ts (hunk 3 above)
Comment (classicalliu) [reply packages/common-scripts/src/sudt.ts]: updated.

**Review by xxuejie — APPROVED** [review]
Let's add a new utility method named `ownerForSudt`: it takes ` FromInfo`, and returns a `Token`. It calculates the SUDT token assuming the FromInfo is used as the owner. After this new method is added we can merge the PR.

**Comment by classicalliu** [comment]
> Let's add a new utility method named `ownerForSudt`: it takes ` FromInfo`, and returns a `Token`. It calculates the SUDT token assuming the FromInfo is used as the owner. After this new method is added we can merge the PR.

Added.

**Review by xxuejie — APPROVED** [review]

**Pull request merged**